### DISCUSSION TITLE
Choppy playback (only happens on very large bitrates and resolution)

### DISCUSSION THREAD

**Discussion by Pcrosser** [aclap-dev/video-downloadhelper#1333] [Help and Q&A]
### Your question

For hls streams with very large bitrate and resolution, after the file is done, it seems to freeze every couple of seconds. (bought the license on Chrome) Only happens for live streams, it's fine if there is an archive. Audio only or smaller resolution/bitrate works fine. I understand I am pressing my PC by downloading 10+ at the same time, but it's fine when I use streamlink, yt-dlp, etc.

At this point I would rather use those programs with command prompts, but I am not adept at gathering the correct commands/cookies to use them. Should I check "do not use coapp when possible"? "Change the variants count"? 

### Addon & CoApp version and details

version: 9.1.0.44
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"f45a29ee76c3437d"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "C:\\Users\\Dragon\\dwhelper",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\Dragon\\dwhelper"
        }
    },
    "concurrent_downloads_max": 32,
    "never_show_no_incognito_msg_again": true,
    "theme": "dark",
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_legacy_ui": false,
    "use_sidebar": true,
    "open_count_store": 1,
    "successfull_dl": 42,
    "never_show_successfull_dl_message": true,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": [
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_2997859199862109"
                    },
                    {
                        "download_result": {
                            "filedir": {
                                "__serializer_tag": "primitive",

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

It's a bit tricky to diagnose problems with livestreams since by the time you post any URL here, it's no longer live.  But if you can give an idea of what authors on what web sites typically post such high bit rate livestreams, maybe somebody else might get lucky & catch a stream to test with while it's still live.  That's pretty much necessary in order to do any sort of useful diagnosis.  We need test cases, otherwise all anybody else can do is shrug their shoulders & move on.

**Reply by GrampaWildWilly** [reply]
The `do not use CoApp` setting won't make any difference.  The CoApp is required for HLS streams, live & otherwise, so you're going to end up using the CoApp regardless of the VDH setting in your case.

The `number of variants` setting affects only how many selections are offered on the VDH menu.  Once you start actually downloading or recording something, that number has no impact.  I would keep that setting at the highest value possible, which is only 10, a number I believe is too low.  The `Max concurrent downloads` setting might have some impact.  That controls how many downloads VDH will attempt at the same time.  Once you've got that many in flight, any further download requests will go into a queue.  When a current download completes, a new download from the queue will get its chance.

But I'm not convinced the level of concurrency will have any effect on your problem.  The true limits on concurrency are the bandwidth you are purchasing from your ISP & whatever throttling the server web sites might be imposing.  It might be helpful if you restrict yourself to a single download/recording at a time & see if the problem occurs.  If it does, then the level of concurrency isn't an issue.  If the problem goes away, then bump the level of concurrency up to 2 & test again.  See if there is a particular level of concurrency below which the problem does not occur & above which it does occur.  We need to try to be scientific about this.

**Comment by Pcrosser** [comment]
Thank you for the rundown! 
I have another problem, sites using this web dev stopped working out of nowhere. 
An example: 
https://nicochannel.jp/unicolony/video/smmnNz9XK7chJ2AV2LqyURZ2

**Reply by GrampaWildWilly** [reply]
I visited that URL.  For some reason, Firefox didn't offer to translate the page so I have no idea what I was looking at.  There was one link that said LIVE so I clicked that.  There were some things on that page that I clicked.  They took me to pages that looked like they might be videos.  But when I clicked on them, nothing happened.  My mouse pointer didn't change to indicate there was anything clickable there.  I suspect it's geoblocking me.

**Reply by Pcrosser** [reply]
I think the site doesn't work on Firefox. Works on Chrome and Edge. Interesting thought about potentially being geoblocked, but I VPN'd to several different places and it plays.

**Reply by Mattmjs** [reply]
It seems that is the case, there is some text in Japanese over the player. I used a site to translate which said the browser
wasn't supported. But if the user agent is changed to chrome or edge then the video plays and detection.
Initiate a download and it fails. Using ffmpeg it's just a 404,  with another tool it is also stumped but there is an error that
suggests it's using encryption. There is just no way it'll work. See the image:

![jp anime](https://github.com/user-attachments/assets/3c113dfc-16b2-45d3-94b1-8c5a4cf815dd)
